Design improvements

ricardoom opened the pull request:
fixes #79

Commit:
:meat_on_bone: removed commented code

Commit:
:meat_on_bone: adding breakpoints to fix alignment of the sub nav moves #79 closer to finished.

Commit:
:meat_on_bone: type improvements for #79
the bottom type size has been reduced by ~15% as per @johneedutton
this is adjusts the type size at the bottom for the reader's position in the book.

aslagle commented:
This happened when I clicked scrolling...
<img width="1429" alt="screen shot 2017-05-02 at 10 30 42 am" src="https://cloud.githubusercontent.com/assets/3409215/25622451/70f23a5a-2f22-11e7-8dd6-845a87d06cc8.png">

aslagle commented:
Also the alignment of the font label message is a little off when you increase the size.
<img width="352" alt="screen shot 2017-05-02 at 10 33 47 am" src="https://cloud.githubusercontent.com/assets/3409215/25622605/e75527ac-2f22-11e7-9eb3-75698925b24c.png">

ricardoom commented:
regarding the first screen shot, I'm not seeing that breaking of the controls to the next line...
regarding the second, i'll fix that. 
I'll pull this pr.

Commit:
Merge branch 'master' into design-improvements

Commit:
:meat_on_bone: commented out all the logic pertaining to resizing the font label
not sure if this the proper approach. @aslagle could you check this a make sure its to you
liking and if so, i'll deleted commmented code, and commit that change.

Commit:
:meat_on_bone: upadted sass to account for a fixed type changer label

ricardoom commented:
@aslagle I made the `.fontSizeLabel` fixed so we dont have such a jumpy experience when resizing the type size...

ricardoom commented:
checks failing due to commented code, not sure how to modify those tests...

Commit:
:meat_on_bone: minor positioning adjustment. lines up better now

aslagle commented on src/BookSettings.ts:
```diff
@@ -210,23 +210,23 @@ export default class BookSettings {
+    //         this.fontSizeButtons["increase"].className = "increase";
+    //     }
+    //
+    //     this.fontSizeLabel.style.fontSize = this.selectedFontSize;
```
you took out a lot more code than you needed to. you could've removed only this last line that sets the font size.

aslagle commented:
@ricardoom I think you could put back most of the code, and then delete the one test that checks if the font size label changes size.

ricardoom commented:
ok. will do.

> On May 2, 2017, at 12:07, Amy Slagle <notifications@github.com> wrote:
> 
> @ricardoom I think you could put back most of the code, and then delete the one test that checks if the font size label changes size.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Commit:
:meat_on_bone: uncommented code as noted by @aslagle.

Commit:
:meat_on_bone: updated book settings so we properly remove the font size increase on the label in the controls.

Commit:
:meat_on_bone: updated tests so we're not looking for a type size increase on the lable as we increase type using the [+] / [-] controls

Commit:
:meat_on_bone: turning overflow scrolling off to prevent the settings control to scroll...

Commit:
:meat_on_bone: put a more specific overflow rule here.
basically overflow no. so now a user wont be able to slide the controls
an improvement would be to get the width of the control exactly to fit the viewport
and then we can remove this rule...

Commit:
:meat_on_bone: for the TOC we _want_ overflow b/c we dont know how long a TOC will be

Commit:
:meat_on_bone: removed the overflow all together

Commit:
Merge branch 'master' into design-improvements

aslagle commented on src/styles/sass/main.scss:
```diff
@@ -293,12 +293,15 @@ a {
 
+.contents-view {
+  overflow: auto;
+}
```
you have overflow:auto in here but overflow: scroll in _toc.scss?

ricardoom replied:
ok. really we want that to get rid of this all together... so TOC has `overflow: scroll` and _settings has `overflow: hidden`

ricardoom replied:
derp. just double declared it as all...

aslagle commented on src/BookSettings.ts:
```diff
@@ -225,7 +225,7 @@ export default class BookSettings {
         }
 
-        this.fontSizeLabel.style.fontSize = this.selectedFontSize;
+        //this.fontSizeLabel.style.fontSize = this.selectedFontSize;
```
could you remove all the commented out lines?

ricardoom replied:
np

Commit:
:meat_on_bone: removed commmented code

Commit:
:meat_on_bone: removed redundant code and settled a long dispute w/ the linter about extra spaces...

aslagle commented on test/BookSettingsTest.ts:
```diff
@@ -219,17 +219,17 @@ describe("BookSettings", () => {
             const decreaseButton = element.querySelector("button[class=decrease]") as HTMLButtonElement;
             const increaseButton = element.querySelector("button[class=increase]") as HTMLButtonElement;
-            const label = element.querySelector(".font-size-label") as HTMLLIElement;
+            //const label = element.querySelector(".font-size-label") as HTMLLIElement;
```
could you remove the commented out lines in the test too?

ricardoom replied:
deeerrrrrrppp. yes.

Commit:
:meat_on_bone: removed more commented lines...

ricardoom commented:
@aslagle can it be dug?

aslagle committed:
Removed a couple more commented out lines.
